test/gtest/common/test_ucc.h

```diff
@@ -10,13 +10,27 @@
 #include <tuple>
 #include <memory>
 
+typedef std::vector<ucc_coll_args_t*> UccCollArgsVec;
+
+class UccCollArgs {
+protected:
+    virtual void init_proc_buf(int nprocs, int rank, uint8_t *buf, size_t len) = 0;
+    virtual int  validate_buf(int nprocs, uint8_t *buf, size_t len) = 0;
```

Comment:
if i will implemente Reduce/allreduce test using this "validate_buf", how will the data check work for floating point types? Are you going to do a uint8_t cast and comparison? it will probably not work.

Reply:
This is intended for alltoall[v] only. A new one should be added for others.